src/common/types/value/value.cpp

```diff
@@ -703,23 +703,58 @@ std::unique_ptr<Value> Value::deserialize(Deserializer& deserializer) {
 
 std::string Value::rdfVariantToString() const {
     auto type = static_cast<LogicalTypeID>(children[0]->val.uint8Val);
+    auto blobData = children[1]->strVal.data();
     switch (type) {
-    case LogicalTypeID::STRING:
+    case LogicalTypeID::STRING: {
         return children[1]->strVal;
+    }
+    case LogicalTypeID::BLOB: {
+        return Blob::toString(Blob::getValue<blob_t>(blobData));
```

Comment:
Instead of using a switch/case here, you can utilize TypeUtils::toString directly:
`TypeUtils::toString(type, blobData, nullptr /* vector */);`